### PR TITLE
Implement editing

### PR DESCRIPTION
**Pull request by parberge**
The edit action can now edit an existing timereport.
#12

### CODE REVIEW

File: chalicelib/action.py [hunk 1]
```diff
@@ -191,13 +190,33 @@ def _edit_action(self):
...
             date_input = "today"
             log.debug(f"Didn't get any params. Setting date to {date_input}")
 
+        try:
+            hours = round(float(self.arguments[2]))
+        except ValueError as error:
+            log.error(f"Failed to parse hours. Error was: {error}")
+            return self.send_response(message="Could not parse hours")
+        except TypeError as error:
+            hours = 8
```
Comment (parberge) [inline chalicelib/action.py]: Unnecessary assignment. Will remove

File: chalicelib/action.py [hunk 2]
```diff
@@ -212,21 +231,22 @@ def _edit_action(self):
 
         event_to_edit = None
         try:
-            event_to_edit = json.loads(self._get_events(date_str=date_input))
+            event_to_edit = read_event(
+                url=self.config["backend_url"], user_id=self.user_id, date=date_input
+            )
         except json.decoder.JSONDecodeError as error:
```
Comment (parberge) [inline chalicelib/action.py]: Hm this exception is probably not going to happen anymore since we don't try to decode json here.
Should I remove the whole try/except?

File: chalicelib/action.py [hunk 3]
```diff
@@ -49,10 +49,12 @@ def perform_action(self):
         """
 
         self.action = self.params[0]
-        log.debug(f"Action is: {self.action}")
+        self.arguments = self.params[1:]
```
Comment (tommymalmqvist) [inline chalicelib/action.py]: What happens if params only has a single value? Like ["help"]

File: chalicelib/action.py (hunk 2 above)
Comment (tommymalmqvist) [reply chalicelib/action.py]: We should check for status 200 on api call

`if event_to_edit.status_code != 200` handle api problem

File: chalicelib/action.py (hunk 3 above)
Comment (parberge) [reply chalicelib/action.py]: `self.arguments` Will then be an empty list

File: chalicelib/action.py (hunk 2 above)
Comment (parberge) [reply chalicelib/action.py]: Yeah but does our API ever return anything else than 200? Probably 500

File: tests/test_action.py [hunk 4]
```diff
@@ -76,3 +76,22 @@ def test_perform_lock():
...
+
+    fake_action.arguments.append("fake_arg_2")
+
+    # Should be valid since we have provided the miniumum and maxiumum amount
+    assert fake_action._valid_number_of_args(min_args=1, max_args=2) is True
+
+    # Should be false since we don't have the minimum amount
+    assert fake_action._valid_number_of_args(min_args=3) is False
+
+    # Should be false since we don't have the maximum amount
```
Comment (kamigerami) [inline tests/test_action.py]: can you add a test for None argument for max_args as well?

File: chalicelib/action.py (hunk 2 above)
Comment (parberge) [reply chalicelib/action.py]: Changed so it checks the `status_code` instead

File: tests/test_action.py (hunk 4 above)
Comment (parberge) [reply tests/test_action.py]: https://github.com/codelabsab/timereport-slack/pull/145/files#diff-0cf678fa971989525e455c58e4d1a529R86 doest that
